odoo/orm/environments.py

```diff
@@ -481,6 +481,28 @@ def get(key, get_context=self.context.get):
             self._cache_key[field] = result
             return result
 
+    @lazy_property
+    def _field_cache(self) -> Callable[[Field], dict[IdType, typing.Any]]:
+        """ Get the read-write cache for field values.
```

Comment:
Make it a real method like `cache_key()`, whose cache is on the environment itself.  This will avoid the cyclic reference issue.

Reply:
With Rémy we found out that with this cached method, you can actually remove the cache of method `cache_key`!  So, instead of memoizing the key to access the cache, you could memoize the result of accessing the cache.

Reply:
Yes, `cache_key` is no longer very useful, but still used for `Field._cache_view`.

Reply:
| function | time |
| -- | -- |
| %timeit self.env._field_cache(self._fields['name']) | 70ns |
| %timeit self._fields['name']._cache_view_for_update(self.env) | 167ns |
| %timeit self._fields['name']._cache_view(self.env) | 165ns |

The view for updates is much more used, because even if you get an attribute, either the view already exists or we will create and populate it; so we call `env._field_cache` directly.
The view that is for reading is used mostly for debugging and backwards compatibiliy.